src/lib/nftClaimer/deploy.ts

```diff
@@ -0,0 +1,100 @@
+import { getAddress } from '@ethersproject/address';
+import { splitSignature } from '@ethersproject/bytes';
+import { Interface } from '@ethersproject/abi';
+import { fetchSpace } from '../../helpers/snapshot';
+import { signer, validateAddresses, validateSpace } from './utils';
+import abi from './deployAbi.json';
+
+const DeployType = {
+  Deploy: [
+    { name: 'implementation', type: 'address' },
+    { name: 'initializer', type: 'bytes' },
+    { name: 'salt', type: 'uint256' }
+  ]
+};
+
+export default async function payload(
+  spaceOwner: string,
+  id: string,
+  maxSupply: number,
+  mintPrice: string,
+  proposerFee: number,
+  salt: string,
+  spaceTreasury: string
+) {
+  const space = await fetchSpace(id);
+  await validateSpace(spaceOwner, space);
+  if (proposerFee < 0 || proposerFee > 100) {
+    throw new Error('proposerFee should be between 0 and 100');
+  }
+  validateAddresses({ spaceOwner, spaceTreasury });
```

Comment:
- Could we move this to line 25 because, we use it in line 26 for `validateSpace`
- Can we pass Array instead of object, which is much easy than doing Object.keys in this function

Reply:
Done. Object is passed to have a descriptive error message, about what is wrong